### DISCUSSION TITLE
Beginner questions about using the el-sub-menu component

### DISCUSSION THREAD

**Discussion by MSTY2003** [element-plus/element-plus#18583] [Q&A] [answered]
I am learning to develop a front-end interface using Vue3 and Element Plus. When following the official documentation to use `el-menu`, I encountered some practical issues. Specifically, when I use `el-menu-item` as a child of `el-menu`, the `el-menu-item` renders correctly. However, if I use `el-sub-menu` as a child of `el-menu` and include some `el-menu-item` as its children, the browser directly renders the subcomponents inside instead of rendering the `el-sub-menu` itself.
Here is my code:
```html
<template>
  <div class="app">
    <el-container style="height: 100%;">
      <el-aside width="20vw" class="aside">
        <el-menu background-color="#304156" text-color="#fff" active-text-color="#ffd04b">
          <el-menu-item index="1">
            <span>Home</span>
          </el-menu-item>
          <el-sub-menu index="2" popper-class="test">
            <template #title>
              <span>About</span>
            </template>
            <el-menu-item index="2-1">item one</el-menu-item>
            <el-menu-item index="2-2">item two</el-menu-item>
            <el-menu-item index="2-3">item three</el-menu-item>
          </el-sub-menu>
        </el-menu>
      </el-aside>
      <el-container>
        <el-header>Header</el-header>
        <el-main>main</el-main>
      </el-container>
    </el-container>
  </div>
</template>

<script>

export default {
  name: 'App',
  components: {
  }
}
</script>

<style lang="scss">
@use '@/assets/colors';

.app {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: colors.$content-background-color;
}

.aside {
  background-color: colors.$secondary-background-black;
}
</style>
```
Here is the actual display effect:
![image](https://github.com/user-attachments/assets/2ce9353a-f822-4822-985b-9bea6332fe21)

I don't know where I made a mistake, hope to get some help.

**Comment by warmthsea** [comment]
Your example looks fine in [Playground Demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IEVsZW1lbnRQbHVzIH0gZnJvbSAnQGVsZW1lbnQtcGx1cy9pY29ucy12dWUnXG5pbXBvcnQgeyB2ZXJzaW9uIGFzIGVwVmVyc2lvbiB9IGZyb20gJ2VsZW1lbnQtcGx1cydcbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuXG5jb25zdCBtc2cgPSByZWYoJ0hlbGxvIFdvcmxkIScpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICAgPGRpdiBjbGFzcz1cImFwcFwiPlxuICAgIDxlbC1jb250YWluZXIgc3R5bGU9XCJoZWlnaHQ6IDEwMCU7XCI+XG4gICAgICA8ZWwtYXNpZGUgd2lkdGg9XCIyMHZ3XCIgY2xhc3M9XCJhc2lkZVwiPlxuICAgICAgICA8ZWwtbWVudSBiYWNrZ3JvdW5kLWNvbG9yPVwiIzMwNDE1NlwiIHRleHQtY29sb3I9XCIjZmZmXCIgYWN0aXZlLXRleHQtY29sb3I9XCIjZmZkMDRiXCI+XG4gICAgICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjFcIj5cbiAgICAgICAgICAgIDxzcGFuPkhvbWU8L3NwYW4+XG4gICAgICAgICAgPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgICAgPGVsLXN1Yi1tZW51IGluZGV4PVwiMlwiIHBvcHBlci1jbGFzcz1cInRlc3RcIj5cbiAgICAgICAgICAgIDx0ZW1wbGF0ZSAjdGl0bGU+XG4gICAgICAgICAgICAgIDxzcGFuPkFib3V0PC9zcGFuPlxuICAgICAgICAgICAgPC90ZW1wbGF0ZT5cbiAgICAgICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIyLTFcIj5pdGVtIG9uZTwvZWwtbWVudS1pdGVtPlxuICAgICAgICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjItMlwiPml0ZW0gdHdvPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgICAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiMi0zXCI+aXRlbSB0aHJlZTwvZWwtbWVudS1pdGVtPlxuICAgICAgICAgIDwvZWwtc3ViLW1lbnU+XG4gICAgICAgIDwvZWwtbWVudT5cbiAgICAgIDwvZWwtYXNpZGU+XG4gICAgICA8ZWwtY29udGFpbmVyPlxuICAgICAgICA8ZWwtaGVhZGVyPkhlYWRlcjwvZWwtaGVhZGVyPlxuICAgICAgICA8ZWwtbWFpbj5tYWluPC9lbC1tYWluPlxuICAgICAgPC9lbC1jb250YWluZXI+XG4gICAgPC9lbC1jb250YWluZXI+XG4gIDwvZGl2PlxuPC90ZW1wbGF0ZT5cbiIsImVsZW1lbnQtcGx1cy5qcyI6ImltcG9ydCBFbGVtZW50UGx1cyBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5cbmxldCBpbnN0YWxsZWQgPSBmYWxzZVxuYXdhaXQgbG9hZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIHNldHVwRWxlbWVudFBsdXMoKSB7XG4gIGlmIChpbnN0YWxsZWQpIHJldHVyblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShFbGVtZW50UGx1cylcbiAgaW5zdGFsbGVkID0gdHJ1ZVxufVxuXG5leHBvcnQgZnVuY3Rpb24gbG9hZFN0eWxlKCkge1xuICBjb25zdCBzdHlsZXMgPSBbJ2h0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vZWxlbWVudC1wbHVzQGxhdGVzdC9kaXN0L2luZGV4LmNzcycsICdodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0BsYXRlc3QvdGhlbWUtY2hhbGsvZGFyay9jc3MtdmFycy5jc3MnXS5tYXAoKHN0eWxlKSA9PiB7XG4gICAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICAgIGxpbmsucmVsID0gJ3N0eWxlc2hlZXQnXG4gICAgICBsaW5rLmhyZWYgPSBzdHlsZVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignZXJyb3InLCByZWplY3QpXG4gICAgICBkb2N1bWVudC5ib2R5LmFwcGVuZChsaW5rKVxuICAgIH0pXG4gIH0pXG4gIHJldHVybiBQcm9taXNlLmFsbFNldHRsZWQoc3R5bGVzKVxufVxuIiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIlBsYXlncm91bmRNYWluLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcydcbnNldHVwRWxlbWVudFBsdXMoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9ydW50aW1lLWRvbUBsYXRlc3QvZGlzdC9ydW50aW1lLWRvbS5lc20tYnJvd3Nlci5qc1wiLFxuICAgIFwiQHZ1ZS9zaGFyZWRcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvc2hhcmVkQGxhdGVzdC9kaXN0L3NoYXJlZC5lc20tYnVuZGxlci5qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9lbGVtZW50LXBsdXNAbGF0ZXN0L2Rpc3QvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9lbGVtZW50LXBsdXNAbGF0ZXN0L1wiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlQDIvZGlzdC9pbmRleC5taW4uanNcIlxuICB9LFxuICBcInNjb3Blc1wiOiB7fVxufSIsIl9vIjp7fX0=)

**Reply by MSTY2003** [reply]
Well, it seems like some problem happened in my development environment. These codes run perfectly fine on the playground, indicating that the code itself is not the issue. Thus, the problem must lie within my development environment. However, I have no clue where to start looking to identify what exactly is wrong.

**Reply by warmthsea** [reply]
> Well, it seems like some problem happened in my development environment. These codes run perfectly fine on the playground, indicating that the code itself is not the issue. Thus, the problem must lie within my development environment. However, I have no clue where to start looking to identify what exactly is wrong.

1、refer to [docs example](https://element-plus.org/en-US/component/menu.html)
2、check you element-plus version and [ensure correct installation](https://element-plus.org/en-US/guide/quickstart.html)
3、check you other code (start with main.ts)

**Reply by MSTY2003** [reply, marked as answer]
The issue has been resolved. The problematic project files had Element Plus installed as a plugin through Vue UI. When I created a new project and installed the relevant dependencies following the official documentation, the previous issues disappeared. Thank you very much for your patient explanations! Does this mean there might be some issues with the Element Plus plugin on Vue UI?

**Reply by warmthsea** [reply]
I don't know what `Vue UI` is, if you want to install `element-plus` please refer to the documentation.
